nucleus/modelci/unit_test.py

```diff
@@ -0,0 +1,142 @@
+"""Unit Tests combine collections of data and evaluation metrics to accelerate model evaluation.
+
+With Model CI Unit Tests, an ML engineer can define a Unit Test from critical
+edge case scenarios that the model must get right (e.g. pedestrians at night),
+and have confidence that they’re always shipping the best model.
+"""
+from dataclasses import dataclass, field
+from typing import List
+
+from ..connection import Connection
+from ..constants import NAME_KEY, SLICE_ID_KEY
+from ..dataset_item import DatasetItem
+from .data_transfer_objects.eval_function import EvaluationCriterion
+from .data_transfer_objects.unit_test_evaluations import GetEvalHistory
+from .data_transfer_objects.unit_test_metric import AddUnitTestMetric
+from .unit_test_evaluation import UnitTestEvaluation
+from .unit_test_metric import UnitTestMetric
+
+DATASET_ITEMS_KEY = "dataset_items"
+
+
+@dataclass
+class UnitTest:
+    """A Unit Test combines a slice and at least one evaluation criterion. A :class:`UnitTest` is not created through
+    the default constructor but using the instructions shown in :class:`ModelCI`. This :class:`UnitTest` class only
+    simplifies the interaction with the unit tests from this SDK.
+
+    Attributes:
+        id (str): The ID of the unit test.
+        connection (Connection): The connection to Nucleus API.
+        name (str): The name of the unit test.
+        slice_id (str): The ID of the associated Nucleus slice.
+    """
+
+    id: str
+    connection: Connection = field(repr=False)
+    name: str = field(init=False)
+    slice_id: str = field(init=False)
+
+    def __post_init__(self):
+        response = self.connection.get(
+            f"modelci/unit_test/{self.id}/info",
+        )
+        self.name = response[NAME_KEY]
+        self.slice_id = response[SLICE_ID_KEY]
+
+    def add_criterion(
+        self, evaluation_criterion: EvaluationCriterion
+    ) -> UnitTestMetric:
+        """Creates and adds a new criteria to the :class:`UnitTest`. ::
+
+            import nucleus
+            client = nucleus.NucleusClient("YOUR_SCALE_API_KEY")
+            unit_test = client.modelci.create_unit_test(
+                "sample_unit_test", "slc_bx86ea222a6g057x4380"
+            )
+
+            e = client.modelci.eval_functions
+            # Assuming a user would like to add all available public evaluation functions as criteria
+            unit_test.add_criterion(
+                e.bbox_iou() > 0.5
```

Comment:
Sorry, can someone explain how this works? I thought that this python syntax would just pass a boolean value to add_criterion. Can the way this works be documented somewhere?

Reply:
Added a follow up ticket for this: https://app.shortcut.com/scaleai/story/314738/document-use-of-pydantic-immutablemodel-and-evaluationcriterion-syntax-for-posterity - I don't feel like I can speak to this very well and we're planning to ship this tonight

FYI @gatli

Reply:
I'll add some docs around this. But the gist of it is that we've overriden the operator methods to define the `EvaluationCriteria`...  like @phil-scale said. 🙂